crates/frontend/src/components/context_form.rs

```diff
@@ -266,8 +344,48 @@ where
         context_ws.update(|v| {
             v.remove(idx);
         });
+        if let Some(removed_dimension) = dimensions
+            .get_value()
+            .iter()
+            .find(|d| d.dimension == d_name)
+        {
+            removed_dimension.dependencies.iter().for_each(|dependent| {
+                dependent_dimension_locks_ws.update(|value| {
+                    value.remove(dependent);
+                });
+            })
+        }
     });
 
+    let get_tool_tip_text = move |condition: StoredValue<Condition>| {
```

Comment:
Can we define the return type with this closure?
```suggestion
    let get_tool_tip_text = move |condition: StoredValue<Condition>| -> String {
```